### PR TITLE
Filtering download links from media-api image response where user does not have permission

### PR DESCRIPTION
**Pull request by phillipbarron**
This update filters download links from media-api response where users should not have access to the download for that image. UI interprets the absence of download links to programmatically determine downloadability. 

Co-authored-by: Ara Cho  <aracho1@users.noreply.github.com>

## What does this change?
No changes to the UI. Media-api response now filters based on user permissions.
Prior to this update, `restrictDownload` was set in `kanuna.config`, this now needs to be moved up to `common.conf` as it configs behaviour both in media-api and the client.

 

## How should a reviewer test this change?
`restrictDownload` config should be set to true in `common.conf`, this enables filtering on download responses from media-api required by the UI to determine whether an image should be downloadable. 

## Tested? Documented?
- [x] locally by committer
- [ ] locally by Guardian reviewer
- [x] on the Guardian's TEST environment
- [ ] relevant documentation added or amended (if needed)

### CODE REVIEW

File: kahuna/public/js/components/gr-downloader/gr-downloader.js [hunk 1]
```diff
@@ -81,36 +71,14 @@ downloader.controller('DownloaderCtrl', [
...
-              ].join('\n');
-
-              $window.alert(message);
-            };
-
-            if (isTooBig) {
-              refuseDownload();
-            } else {
-              createDownload();
-            }
```
Comment (andrew-nowak) [inline kahuna/public/js/components/gr-downloader/gr-downloader.js:111]: nice cleanup!

File: media-api/app/lib/ImageExtras.scala [hunk 2]
```diff
@@ -38,27 +38,43 @@ object ImageExtras {
...
+  }
+
+  def validityMap(image: Image, withWritePermission: Boolean)(
+    implicit cost: CostCalculator, quotas: UsageQuota): ValidMap = {
+    validationMap(image, withWritePermission, isImageValidation = true)
+  }
+
+  def downloadableMap(image: Image, withWritePermission: Boolean)(
+    implicit cost: CostCalculator, quotas: UsageQuota): ValidMap = {
+    validationMap(image, withWritePermission, isImageValidation = false)
```
Comment (andrew-nowak) [inline media-api/app/lib/ImageExtras.scala:74]: and same problem here

**Review by andrew-nowak — COMMENTED** [review]
looks great, thanks for all of the cleanups (deleting dead code, fixing indents, etc.) - really appreciated!

just a couple checks then will lgtm!

File: media-api/app/lib/ImageExtras.scala [hunk 3]
```diff
@@ -38,27 +38,43 @@ object ImageExtras {
...
+      )
+    } else {
+      baseValidationMap
+    }
+  }
+
+  def validityMap(image: Image, withWritePermission: Boolean)(
+    implicit cost: CostCalculator, quotas: UsageQuota): ValidMap = {
+    validationMap(image, withWritePermission, isImageValidation = true)
+  }
```
[the remark this answers is not in the document]
Comment (phillipbarron) [reply media-api/app/lib/ImageExtras.scala:70]: Yeah 100% agree - thank you 👍

**Review by andrew-nowak — APPROVED** [review]

**Pull request merged**